src/hlsBinaries.ts

```diff
@@ -206,44 +215,54 @@ export async function findHaskellLanguageServer(
         fs.mkdirSync(storagePath);
     }
 
-    const manageHLS = workspace.getConfiguration('haskell').get('manageHLS') as boolean;
+    if (manageHLS === null) { // plugin needs initialization
+      const promptMessage =
+          'How do you want the extension to manage/discover HLS?';
+
+      const decision = await window.showInformationMessage(promptMessage, 'system ghcup (recommended)', 'internal ghcup', 'PATH') || null;
+      if (decision === 'system ghcup (recommended)') {
+        manageHLS = 'system-ghcup';
+      } else if (decision === 'internal ghcup') {
+        manageHLS = 'internal-ghcup';
+      } else if (decision === 'PATH') {
+        manageHLS = 'PATH';
+      }
+      if (manageHLS !== null) {
+        workspace.getConfiguration('haskell').update('manageHLS', manageHLS, ConfigurationTarget.Global);
+      }
+    }
 
-    if (!manageHLS) {
-        const wrapper = findHLSinPATH(context, logger, folder);
-        if (!wrapper) {
-            const msg = 'Could not find a HLS binary! Consider installing HLS via ghcup or set "haskell.manageHLS" to true';
-            window.showErrorMessage(msg);
-            throw new Error(msg);
+    if (manageHLS === 'PATH' || manageHLS === null) {
+        if (workspace.getConfiguration('haskell').get('serverExecutablePath') as string !== '') {
+          return findServerExecutable(context, logger, folder);
```

Comment:
Currently we only pick up on `serverExecutablePath` if `manageHLS === PATH`. Do we want different behavior?